src/main/java/vg/civcraft/mc/civmodcore/serialization/NBTSerializable.java

```diff
@@ -0,0 +1,33 @@
+package vg.civcraft.mc.civmodcore.serialization;
+
+import vg.civcraft.mc.civmodcore.api.NBTCompound;
+
+/**
+ * Interface that grants a class access to {@link NBTSerialization NBTSerialization's} process of converting a class to
+ * and from an NBTCompound.
+ * */
+public interface NBTSerializable {
+
+	/**
+	 * Serialises a class into an NBTCompound.
+	 *
+	 * @param nbt The NBTCompound to serialise into, which <i>should</i> NEVER be null, so feel free to throw an
+	 * {@link NBTSerializationException} if it is. You can generally assume that the nbt compound is new and therefore
+	 * empty, but you may wish to check or manually {@link NBTCompound#clear() empty it}, though the latter may cause
+	 * other issues.
+	 *
+	 * @throws NBTSerializationException This is thrown if the implementation has a fatal error serializing.
+	 * */
+	void serialize(NBTCompound nbt) throws NBTSerializationException;
```

Comment:
If you assume the used NBTCompound to be empty, why not just create a new one in this method and return it?

Reply:
> You can generally assume that the nbt compound is new and therefore empty

Working with a given NBTCompound rather than creating one's own allows classes to serialise themselves onto already populated compounds if the use case so warrants it. Whereas the method always creating its own compound and returning it assumes that you always, *always* want one compound to match one instance, which assumes that NBTSerializable will always be used just for NBTSerialization, which is not necessarily true.